compiler/rustc_middle/src/query/mod.rs

```diff
@@ -716,6 +716,10 @@ rustc_queries! {
                 "const-evaluating + checking `{}`",
                 key.value.display(tcx)
             }
+            cache_on_disk_if(_, opt_result) {
+                // Only store results without errors
```

Comment:
Why do we only want to store results without errors? Or is this a common thing to do?

Reply:
I think the original motivation was that we can't encode the error, but since today that's just an `ErrorHandled` and not the error itself anymore, we can probably start writing the entire thing to disk again.

Reply:
Hm, @oli-obk did you intend to update this to remove this if here?